Add Jazzy job configurations.

nuclearsandwich opened the pull request:
Adds CI, documentation, release source, and binary jobs for ROS 2 Jazzy. Based on the ROS 2 Rolling configurations with: `sed -i 's/rolling/jazzy/g'`.

nuclearsandwich committed:
Add Jazzy job configurations.

Adds CI, documentation, release source, and binary jobs for ROS 2 Jazzy.
Based on the ROS 2 Rolling configurations with:
`sed -i 's/rolling/jazzy/g'`.

nuclearsandwich committed:
Remove file that snuck in via a dirty workspace.

clalancette commented on index.yaml:
```diff
@@ -61,6 +61,35 @@ distributions:
+      nightly-cross-vendor-connext-cyclonedds: jazzy/ci-nightly-cross-vendor-connext-cyclonedds.yaml
+      nightly-cross-vendor-connext-fastrtps: jazzy/ci-nightly-cross-vendor-connext-fastrtps.yaml
+      nightly-cross-vendor-connext-fastrtps-dynamic: jazzy/ci-nightly-cross-vendor-connext-fastrtps-dynamic.yaml
+      nightly-cross-vendor-cyclonedds-fastrtps: rollingjazzyci-nightly-cross-vendor-cyclonedds-fastrtps.yaml
```
```suggestion
      nightly-cross-vendor-cyclonedds-fastrtps: jazzy/ci-nightly-cross-vendor-cyclonedds-fastrtps.yaml
```

nuclearsandwich replied:
b9692e9c6fb8020ee2556cf490f6b009ff335f26

nuclearsandwich committed:
Fix misapplied edit.

This is why I use awk and sed so I can only make these silly human
mistakes at a grand scale.

clalancette reviewed (approved):
Looks good to me, thanks for doing this!

nuclearsandwich commented:
Commands for deploying jazzy to the buildfarm:

```
# Dry-run to make sure things is good
❯ generate_all_jobs.py https://raw.githubusercontent.com/ros2/ros_buildfarm_config/ros2/index.yaml --ros-distro-names jazzy
# Review the output ...
❯ generate_all_jobs.py https://raw.githubusercontent.com/ros2/ros_buildfarm_config/ros2/index.yaml --ros-distro-names jazzy --commit
...
❯ generate_release_trigger_upload_jobs.py https://raw.githubusercontent.com/ros2/ros_buildfarm_config/ros2/index.yaml
```

ros-discourse commented:
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/jazzy-pre-release-is-ready-for-the-tutorial-party/37472/1
